src/aggregation/agg_req.rs

```diff
@@ -110,9 +110,14 @@ impl BucketAggregationInternal {
             _ => None,
         }
     }
-    pub(crate) fn as_histogram(&self) -> Option<&HistogramAggregation> {
+    pub(crate) fn as_histogram(&self) -> Option<HistogramAggregation> {
         match &self.bucket_agg {
-            BucketAggregationType::Histogram(histogram) => Some(histogram),
+            BucketAggregationType::Histogram(histogram) => Some(histogram.clone()),
+            BucketAggregationType::DateHistogram(histogram) => Some(
+                histogram
+                    .to_histogram_req()
+                    .expect("could not convert date histogram req to histogram req"),
```

Comment:
I have no clue if this expect is justified or not... O_o

Can you remove it?

Expect is ok if it is reasonably obvious that it should never happen.

A good rule of thumb...
It is not ok if it depends on code running in a different file or a different computer to do "the right thing".

If it is something justified (it should never happen) trivially by the surrounding code, then expect is perfectly fine.

Reply:
The method can fail in general, because of the parse into microseconds conversion, but not at this time, since the conversion was already done before.

I think I can change it to return `Result<Option>`